internal/common/baseclient/baseclient.go

```diff
@@ -131,18 +130,16 @@ func (c *BaseClient) GetVoteForRule(ruleName string) bool {
 
 // GetVoteForElection returns the client's Borda vote for the role to be elected.
 // COMPULSORY: use opinion formation to decide a rank for islands for the role
-func (c *BaseClient) GetVoteForElection(roleToElect shared.Role) []shared.ClientID {
+func (c *BaseClient) VoteForElection(roleToElect shared.Role, candidateList []shared.ClientID) []shared.ClientID {
 	// Done ;)
 	// Get all alive islands
-	aliveClients := rules.VariableMap[rules.IslandsAlive]
-	// Convert to ClientID type and place into unordered map
-	aliveClientIDs := map[int]shared.ClientID{}
-	for i, v := range aliveClients.Values {
-		aliveClientIDs[i] = shared.ClientID(int(v))
+	candidates := map[int]shared.ClientID{}
+	for i := 0; i < len(candidateList); i++ {
+		candidates[i] = candidateList[i]
 	}
 	// Recombine map, in shuffled order
 	var returnList []shared.ClientID
-	for _, v := range aliveClientIDs {
+	for _, v := range candidates {
 		returnList = append(returnList, v)
```

Comment:
And then copying again from `candidates` to `returnList`

Reply:
It is to get a list of candidate IDs in shuffled order by ranging map.